testing/src/main/scala/org/http4s/testing/ArbitraryInstances.scala

```diff
@@ -715,9 +729,100 @@ trait ArbitraryInstances {
       f <- arbitrary[A => F[Entity[F]]]
       hs <- arbitrary[Headers]
     } yield EntityEncoder.encodeBy(hs)(f))
-}
 
-object ArbitraryInstances extends ArbitraryInstances {
-  // This were introduced after .0 and need to be kept out of the
-  // trait.  We can move them back into the trait in the next .0.
+  implicit def arbitraryEntityDecoder[F[_], A](implicit 
+    F: Effect[F],
+    ec: TestContext,
+    g: Arbitrary[DecodeResult[F, A]]
+  ) =
+    Arbitrary(for {
+      f <- arbitrary[(Message[F], Boolean) => DecodeResult[F, A]]
+      mrs <- arbitrary[Set[MediaRange]]
+    } yield new EntityDecoder[F, A] {
+      def decode(msg: Message[F], strict: Boolean): DecodeResult[F, A] = f(msg, strict)
+      def consumes = mrs
+    })
+
+  implicit def cogenMessage[F[_]](implicit F: Effect[F], ec: TestContext): Cogen[Message[F]] =
+    Cogen[(Headers, EntityBody[F])].contramap(m => (m.headers, m.body))
+
+  implicit def cogenHeaders: Cogen[Headers] =
+    Cogen[List[Header]].contramap(_.toList)
+
+  implicit def cogenHeader: Cogen[Header] =
+    Cogen[(CaseInsensitiveString, String)].contramap(h => (h.name, h.value))
+
+  implicit def arbitraryDecodeFailure: Arbitrary[DecodeFailure] =
```

Comment:
Incomplete.  `DecodeFailure` is an open hierarchy.  Which I dearly wish it weren't.  If we do complete this, then we have a problem in the Cogen instance.